satpy/writers/ninjogeotiff.py

```diff
@@ -0,0 +1,214 @@
+# Copyright (c) 2021- Satpy developers
+#
+# This file is part of satpy.
+#
+# satpy is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+"""Writer for GeoTIFF images with tags for the NinJo visualization tool.
+
+Since NinJo version 7 (released spring 2022), NinJo is able to read standard
+GeoTIFF images, with required metadata encoded as a set of XML tags in the
+GDALMetadata TIFF tag.  Each of the XML tags must be prepended with 'NINJO_'.
+"""
+
+from .geotiff import GeoTIFFWriter
+
+
+class NinJoGeoTIFFWriter(GeoTIFFWriter):
+    """Writer for GeoTIFFs with NinJo tags."""
+
+    def save_dataset(self, dataset, ninjo_tags):
+        """Save dataset along with NinJo tags.
+
+        Save dataset along with NinJo tags.  Interface as for GeoTIFF, except
+        it takes an additional keyword argument ninjo_tags.  Those tags will be
+        prepended with ninjo_ and added as GDALMetaData.
```

Comment:
Oooohhh, our previous conversation makes more sense now. I didn't realize you wanted the final tags to have the `ninjo_` prefix in the geotiff metadata. I am much less resistant to the idea of requiring users to put a `ninjo_` prefix on their tags when they pass them to the writer.